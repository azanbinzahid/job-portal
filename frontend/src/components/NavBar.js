import React from 'react'
import {connect, useSelector} from 'react-redux'
import {Navbar, Nav} from 'react-bootstrap'
import { LinkContainer } from 'react-router-bootstrap';
import styled from 'styled-components'
import {logUserOut} from 'redux/actions'




const Item = styled.h4``;
const Title = styled.h2`
    font-weight: bold;
    color: blue;
`;


const NavItem = (toPath, menuName) => {
    return (
        <LinkContainer to={toPath}>
            <Nav.Link>
                <Item>
                    {menuName}
                </Item>
            </Nav.Link>
        </LinkContainer>
    )
}


const NavBar = (props) => {

    const userReducer = useSelector(state => state.userReducer)
    
    return (
        <Navbar bg="light" expand="lg">
        <LinkContainer to="/">
        <Navbar.Brand> <Title> Job Portal </Title></Navbar.Brand>
        </LinkContainer>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
            <Nav className="ml-auto">
                {NavItem("/", "Home")}
                {
                        !userReducer.loggedIn ? 
                            <>
                            {NavItem("/login", "Login")}
                            {NavItem("/signup", "SignUp")}
                            </>
                         : 
                            <>
                            {NavItem("/jobs", "Jobs")}
                            <Nav.Link onClick={props.logUserOut}> <Item>Logout</Item> </Nav.Link>
                            </>

                }

            </Nav>
        </Navbar.Collapse>
        </Navbar>  
    )
}



const mapDispatchToProps = (dispatch) => {
    return {
        logUserOut: () => dispatch(logUserOut()),
    }
  }
  
export default connect(null, mapDispatchToProps)(NavBar);
