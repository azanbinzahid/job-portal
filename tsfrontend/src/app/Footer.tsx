import React from "react";
import { Jumbotron } from "react-bootstrap";

export const Footer = () => {
  return (
    <Jumbotron<React.ElementType> className="m-0 mt-5 p-5" align="center">
      Â© 2020 Job Portal. All Rights Reserved.
    </Jumbotron>
  );
};

export default Footer;
