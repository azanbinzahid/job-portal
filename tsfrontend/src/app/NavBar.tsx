import React, { FC } from "react";
import { connect } from "react-redux";
import { Navbar, Nav } from "react-bootstrap";
import { LinkContainer } from "react-router-bootstrap";
import styled from "styled-components";
import { logUserOut } from "redux/actions";
import { RootState } from "redux/reducers";

const Title = styled.h3`
  display: inline;
  font-weight: bold;
  color: #1d83d4;
  vertical-align: bottom;
`;

const NavItem = (toPath: string, menuName: string) => {
  return (
    <LinkContainer to={toPath}>
      <Nav.Link>
        <h4>{menuName}</h4>
      </Nav.Link>
    </LinkContainer>
  );
};

type Props = {
  isLogged: boolean;
  logUserOut: () => void;
};

export const NavBar: FC<Props> = (props) => {
  return (
    <Navbar bg="light" expand="lg">
      <LinkContainer to="/">
        <Navbar.Brand>
          <img
            alt=""
            src="https://image.flaticon.com/icons/svg/3232/3232975.svg"
            width="50"
            height="50"
            className="d-inline-block align-top"
          />
          <Title> Job Portal</Title>
        </Navbar.Brand>
      </LinkContainer>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="ml-auto">
          {!props.isLogged ? (
            <>
              {NavItem("/", "Home")}
              {NavItem("/jobs", "Jobs")}
              {NavItem("/login", "Login")}
              {NavItem("/signup", "SignUp")}
            </>
          ) : (
            <>
              {NavItem("/", "Home")}
              {NavItem("/jobs", "Jobs")}
              {NavItem("/profile", "Profile")}
              <Nav.Link onClick={props.logUserOut}>
                <h4>Logout</h4>
              </Nav.Link>
            </>
          )}
        </Nav>
      </Navbar.Collapse>
    </Navbar>
  );
};

const mapStateToProps = (state: RootState) => ({
  isLogged: state.userReducer.loggedIn,
});

const mapDispatchToProps = {
  logUserOut,
};

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
