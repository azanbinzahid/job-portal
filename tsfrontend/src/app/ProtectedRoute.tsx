import React, { FC } from "react";
import { connect } from "react-redux";
import {
  Redirect,
  Route,
  RouteComponentProps,
  RouteProps,
} from "react-router-dom";
import { Spinner, Container } from "react-bootstrap";
import { RootState } from "redux/reducers";

interface PrivateRouteProps extends RouteProps {
  isLogged: boolean;
  loadded: boolean;
  path: string;
  component:
    | React.ComponentType<RouteComponentProps<any>>
    | React.ComponentType<any>
    | any;
}

export const ProtectedRoute: FC<PrivateRouteProps> = (props) => {
  const { component: Component, ...rest } = props;

  if (!props.loadded) {
    return (
      <Container<React.ElementType> align="center" className="p-5">
        <Spinner animation="border" />
      </Container>
    );
  }

  return props.isLogged ? (
    <Route {...rest} render={(props) => <Component {...props} />} />
  ) : (
    <Redirect
      to={{
        pathname: "/login",
      }}
    />
  );
};

const mapStateToProps = (state: RootState) => ({
  isLogged: state.userReducer.loggedIn,
  loadded: state.userReducer.loadded,
});

export default connect(mapStateToProps)(ProtectedRoute);
