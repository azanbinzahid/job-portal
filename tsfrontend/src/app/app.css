/* @media (width: 999px) {
  .card-columns {
    column-count: 2;
  }
} */
