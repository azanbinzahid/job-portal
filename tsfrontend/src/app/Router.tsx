import React, { useEffect, FC } from "react";
import { connect } from "react-redux";
import { BrowserRouter, Switch, Route } from "react-router-dom";
import { autoLogin, fetchJobs } from "redux/actions";
import Home from "home/Home";
import Login from "auth/Login";
import Signup from "auth/Signup";
import NavBar from "app/NavBar";
import JobList from "jobs/JobList";
import JobDetailed from "jobs/JobDetailed";
import Profile from "profile/Profile";
import ProtectedRoute from "app/ProtectedRoute";

type Props = {
  autoLogin: () => void;
  fetchJobs: () => void;
};
export const Router: FC<Props> = (props) => {
  useEffect(() => {
    props.autoLogin();
    props.fetchJobs();
  }, [props]);

  return (
    <BrowserRouter>
      <NavBar />
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/login" component={Login} />
        <Route path="/signup" component={Signup} />
        <ProtectedRoute path="/jobs/:jobId" component={JobDetailed} />
        <ProtectedRoute path="/profile" component={Profile} />
        <Route path="/jobs" component={JobList} />
        <Route component={Home} />
      </Switch>
    </BrowserRouter>
  );
};

const mapDispatchToProps = {
  autoLogin,
  fetchJobs,
};

export default connect(null, mapDispatchToProps)(Router);
