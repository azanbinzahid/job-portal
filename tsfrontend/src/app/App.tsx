import React from "react";
import Router from "app/Router";
import { Provider } from "react-redux";
import store from "redux/store";

export default function App() {
  return (
    <Provider store={store}>
      <Router />
    </Provider>
  );
}
