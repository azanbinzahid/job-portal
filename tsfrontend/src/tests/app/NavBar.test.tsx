import React from "react";
import { shallow } from "enzyme";
import { NavBar } from "app/NavBar";
test("renders the component", () => {
  const component = shallow(<NavBar isLogged={true} logUserOut={jest.fn()} />);
  expect(component).toMatchSnapshot();
});
test("renders the component", () => {
  const component = shallow(<NavBar isLogged={false} logUserOut={jest.fn()} />);
  expect(component).toMatchSnapshot();
});
