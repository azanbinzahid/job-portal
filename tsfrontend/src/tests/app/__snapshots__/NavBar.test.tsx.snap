// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  variant="light"
>
  <withRouter(LinkContainer)
    to="/"
  >
    <NavbarBrand>
      <img
        alt=""
        className="d-inline-block align-top"
        height="50"
        src="https://image.flaticon.com/icons/svg/3232/3232975.svg"
        width="50"
      />
      <styled.h3>
         Job Portal
      </styled.h3>
    </NavbarBrand>
  </withRouter(LinkContainer)>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="basic-navbar-nav"
  >
    <Nav
      className="ml-auto"
      fill={false}
      justify={false}
    >
      <Connect(Component) />
      <withRouter(LinkContainer)
        to="/"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <h4>
            Home
          </h4>
        </NavLink>
      </withRouter(LinkContainer)>
      <p />
      <withRouter(LinkContainer)
        to="/jobs"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <h4>
            Jobs
          </h4>
        </NavLink>
      </withRouter(LinkContainer)>
      <p>
        <Badge
          pill={false}
          variant="primary"
        >
          New
        </Badge>
      </p>
      <withRouter(LinkContainer)
        to="/profile"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <h4>
            Profile
          </h4>
        </NavLink>
      </withRouter(LinkContainer)>
      <p />
      <NavLink
        as={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "SafeAnchor",
            "render": [Function],
          }
        }
        disabled={false}
        onClick={[MockFunction]}
      >
        <h4>
          Logout
        </h4>
      </NavLink>
    </Nav>
  </NavbarCollapse>
</Navbar>
`;

exports[`renders the component 2`] = `
<Navbar
  bg="light"
  collapseOnSelect={false}
  expand="lg"
  variant="light"
>
  <withRouter(LinkContainer)
    to="/"
  >
    <NavbarBrand>
      <img
        alt=""
        className="d-inline-block align-top"
        height="50"
        src="https://image.flaticon.com/icons/svg/3232/3232975.svg"
        width="50"
      />
      <styled.h3>
         Job Portal
      </styled.h3>
    </NavbarBrand>
  </withRouter(LinkContainer)>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="basic-navbar-nav"
  >
    <Nav
      className="ml-auto"
      fill={false}
      justify={false}
    >
      <Connect(Component) />
      <withRouter(LinkContainer)
        to="/"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <h4>
            Home
          </h4>
        </NavLink>
      </withRouter(LinkContainer)>
      <p />
      <withRouter(LinkContainer)
        to="/jobs"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <h4>
            Jobs
          </h4>
        </NavLink>
      </withRouter(LinkContainer)>
      <p>
        <Badge
          pill={false}
          variant="primary"
        >
          New
        </Badge>
      </p>
      <withRouter(LinkContainer)
        to="/login"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <h4>
            Login
          </h4>
        </NavLink>
      </withRouter(LinkContainer)>
      <p />
      <withRouter(LinkContainer)
        to="/signup"
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          disabled={false}
        >
          <h4>
            SignUp
          </h4>
        </NavLink>
      </withRouter(LinkContainer)>
      <p />
    </Nav>
  </NavbarCollapse>
</Navbar>
`;
